joomla/joomla-cms
# Use a sensible default if none is found

### Summary of Changes
When you have a module installed but the module manifest does not have the `client` attribute set, uninstalling this module crashes:
![image](https://user-images.githubusercontent.com/359377/149615714-9a668a72-5114-4c91-9543-b7a49ec676c5.png)

The same thing happens when you install the module but when you refresh the page it tells you the module installation was fine.

So the change here is to set a sensible default of `site` as that is where most modules are installed.

This was tested on PHP 8.1

### Testing Instructions
1. Go to System -> Global Configuration
2. Click on the Server tab
3. Set Error reporting to Maximum
4. Click on Save & Close
5. Go to System -> Install -> Extensions
6. Install the attached module
[mod_test.zip](https://github.com/joomla/joomla-cms/files/7874622/mod_test.zip)
7. See that you get the error as shown above but the module is installed
8. Uninstall the module
9. See that you get the error as shown above but the module is uninstalled
10. Apply the patch
11. Install the module again
12. Notice that there is no error now
13. Uninstall the module
14. Notice that there is no error now


### Actual result BEFORE applying this Pull Request
The module does install but the user faces a fatal error


### Expected result AFTER applying this Pull Request
The module is installed but no fatal error is shown


### Documentation Changes Required
None
